src/modm/board/disco_f469ni/board_dsi_nt35510.cpp

```diff
@@ -0,0 +1,220 @@
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+#include <modm/platform/clock/rcc.hpp>
+
+extern void
+nt35510_init(uint8_t);
+
+// ---------------------------------- DISPLAY ----------------------------------
+void
+board_initialize_display(uint8_t ColorCoding)
+{
+	using namespace modm::platform;
+	if (Rcc::isEnabled<Peripheral::Ltdc>()) return;
+	// Enable clock to LTDC, DSI interface
+	Rcc::enable<Peripheral::Ltdc>();
+	Rcc::enable<Peripheral::Dsihost>();
+
+	{
+		// Expanded `HAL_DSI_Init()`:
+		// Enable regulator
+		DSI->WRPCR = DSI_WRPCR_REGEN;
+		// Wait until stable
+		for (int t = 1'024; not (DSI->WISR & DSI_WISR_RRS) and t; t--) {
+			modm::delay_ms(1);
+		}
+		// Set up PLL and enable it
+		DSI->WRPCR |= (0 << 16) | (2 << 11) | (125 << 2) | DSI_WRPCR_PLLEN;
+		// Wait until stable
+		for (int t = 1'024; not (DSI->WISR & DSI_WISR_PLLLS) and t; t--) {
+			modm::delay_ms(1);
+		}
+		// D-PHY clock and digital enable
+		DSI->PCTLR = DSI_PCTLR_CKE | DSI_PCTLR_DEN;
+		// Clock lane configuration
+		DSI->CLCR = DSI_CLCR_DPCC;
+		// Configure the number of active data lanes
+		DSI->PCONFR = 1;
+		// Set the TX escape clock division factor
+		DSI->CCR = 4;
+		// Calculate the bit period in high-speed mode in unit of 0.25 ns (UIX4)
+		// The equation is : UIX4 = IntegerPart( (1000/F_PHY_Mhz) * 4 )
+		// Where : F_PHY_Mhz = (NDIV * HSE_Mhz) / (IDF * ODF)
+		// Set the bit period in high-speed mode
+		DSI->WPCR[0] = 8;
+		// Disable all error interrupts and reset the Error Mask
+		DSI->IER[0] = 0;
+		DSI->IER[1] = 0;
+	}
+
+	constexpr uint32_t VSA = 2;
+	constexpr uint32_t VBP = 34;
+	constexpr uint32_t VFP = 34;
+
+	constexpr uint32_t HSA = 120;
+	constexpr uint32_t HBP = 150;
+	constexpr uint32_t HFP = 150;
+
+	constexpr uint32_t HACT = 800;
+	constexpr uint32_t VACT = 480;
+	const uint8_t pixel_size = (ColorCoding == 0) ? sizeof(uint32_t) : sizeof(uint16_t);
+	constexpr float ClockRatio = 62500.f / 27429;
+
+	{
+		// Expanded `HAL_DSI_ConfigVideoMode()`
+		// Select video mode by resetting CMDM and DSIM bits
+		DSI->MCR = 0;
+		DSI->WCFGR = 0;
+		// Configure the video mode transmission type
+		DSI->VMCR = 2;
+		// Configure the video packet size
+		DSI->VPCR = HACT;
+		// Set the chunks number to be transmitted through the DSI link
+		DSI->VCCR = 0;
+		// Set the size of the null packet
+		DSI->VNPCR = 0xFFF;
+		// Select the virtual channel for the LTDC interface traffic
+		DSI->LVCIDR = 0;
+		// Configure the polarity of control signals
+		DSI->LPCR = 0;
+		// Select the color coding for the host
+		DSI->LCOLCR = ColorCoding;
+		// Select the color coding for the wrapper
+		DSI->WCFGR = ColorCoding << 1;
+		// DSI_VSYNC_RISING
+		DSI->WCFGR = DSI_WCFGR_VSPOL;
```

Comment:
This looks a bit odd. In Line 77, `WCFGR` is set to 0, then to `ColorCoding <<1`  and then to `DSI_WCFGR_VSPOL`.

Reply:
The second assignment should probably be `DSI->WCFGR |= DSI_WCFGR_VSPOL;`.

Thanks for catching this and notifying here!